backend/src/test/kotlin/com/group7/artshare/controller/EventControllerTest.kt

```diff
@@ -0,0 +1,57 @@
+package com.group7.artshare.controller
+
+import com.group7.artshare.entity.*
+import com.group7.artshare.service.EventService
+import com.group7.artshare.service.JwtService
+import org.junit.jupiter.api.Test
+
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.kotlin.whenever
+
+@ExtendWith(MockitoExtension::class)
+internal class EventControllerTest {
+
+    @InjectMocks
+    lateinit var eventController: EventController
+
+    @Mock
+    lateinit var eventService: EventService
+
+    @Mock
+    lateinit var jwtService: JwtService
+
+    @Test
+    fun participateAnEvent() {
+        val authorizationHeader = "Bearer token"
+        val user = RegisteredUser(AccountInfo("email","username", "password"),setOf(Authority("ARTIST")))
+        whenever(jwtService.getUserFromAuthorizationHeader(authorizationHeader)).thenReturn(user)
+        val event = PhysicalExhibition()
+        Mockito.`when`(eventService.participateAnEvent(event.id,user)).then {
+            event.participants.add(user)
+            event
+        }
+        val result = eventController.participateAnEvent(event.id,authorizationHeader)
+        assertEquals(1, result.participantUsernames.size)
+        assertEquals(user.username, result.participantUsernames[0])
+    }
+
+    @Test
+    fun unparticipateAnEvent() {
```

Comment:
I think the thing you are mocking is missing the point. What I suggest and see is as below:
you should mock physicalExhibitionRepository.findByIdOrNull given the id you set for the event to return the event you provide in **given**. Then rest will follow up. You are in current version directly doing the actions you expect the service you are testing to do in the second "when". Thus, it is shadowing possible mishaps to occur.

Reply:
There is a problem related with repository mocking on Kotlin. You can see it on here: http://www.kenansevindik.com/en/mockito-and-extension-functions-in-kotlin/

They stated that "The reason for that problem is the way how extension functions are implemented in Kotlin. They are simply defined as static methods. Therefore, when you invoke the extension function through the mock object, the static method is being invoked instead of the trained one in your mock instance. You can find more information about it [here](https://kotlinlang.org/docs/extensions.html#extensions-are-resolved-statically).

So what is the solution? Unfortunately, there seems no solution for this in Mockito, yet."  

That's why this is the best I can do.